aten/src/ATen/TensorMeta.h

```diff
@@ -62,14 +77,58 @@ namespace impl {
 //
 // A notable subclass of this interface is TensorIteratorBase.
 struct TORCH_API MetaBase {
-  virtual void set_output(int64_t output_idx, IntArrayRef sizes, IntArrayRef strides, TensorOptions options, DimnameList names) = 0;
+  virtual void set_output(int64_t output_idx, IntArrayRef sizes, IntArrayRef strides, TensorOptions options, DimnameList names) {
+    set_output_raw_strided(output_idx, sizes, strides, options, names);
+  }
+
   virtual const Tensor& maybe_get_output(int64_t output_idx) = 0;
   void set_output(IntArrayRef sizes, TensorOptions options) {
     set_output(0, sizes, {}, options, {});
   }
   void set_output(int64_t output_idx, IntArrayRef sizes, TensorOptions options) {
     set_output(output_idx, sizes, {}, options, {});
   }
+
+  // See: https://github.com/pytorch/pytorch/issues/69813
+  // Whenever defining the output properties in the META function of a structured
+  // kernel (what was usually done with `set_output`), use one of these 3 variants,
+  // instead.
+  //
+  // Use this function whenever the kernel requires specific strides for the output.
+  // If `strides` does not match the given output strides, proxy outputs will be
+  // created and passed to the IMPL function.
+  virtual void set_output_strided(
+      int64_t output_idx,
+      IntArrayRef sizes,
+      IntArrayRef strides,
+      TensorOptions options,
+      DimnameList names = {}) {
+    TORCH_INTERNAL_ASSERT(false, "set_output_strided not implemented.");
+  }
+
+  // Use this function whenever the kernel knows how to handle arbitrary strided outputs.
+  // This function has the same behavior as the old `set_output`: it will only
+  // re-stride if the given output was resized.
+  virtual void set_output_raw_strided(
+      int64_t output_idx,
+      IntArrayRef sizes,
+      IntArrayRef strides,
```

Comment:
I guess I'd suggest naming this something like `strides_hint` to indicate that you won't get this strides if an explicit out tensor is passed